pkg/ccl/changefeedccl/compression.go

```diff
@@ -45,11 +133,79 @@ func newCompressionCodec(
 	switch algo {
 	case sinkCompressionGzip:
 		if useFastGzip.Get(sv) {
-			return pgzip.NewWriterLevel(dest, pgzip.DefaultCompression)
+			pooled := fastGzipEncoderPool.Get()
+			if pooled != nil {
+				encoder := pooled.(*encWrapper[*pgzip.Writer])
+				encoder.Reset(dest)
+				return encoder, nil
+			}
+			encoder, err := pgzip.NewWriterLevel(dest, pgzip.DefaultCompression)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to create pgzip encoder")
+			}
+			return &encWrapper[*pgzip.Writer]{encoder, &fastGzipEncoderPool}, nil
```

Comment:
nit: use field names in this struct literal please

Reply:
Fixed